VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java

```diff
@@ -291,16 +291,17 @@ public Fragment getItem(int pos) {
             //FIXME - for the video, let's ignore studentViewMultiDevice for now
             if (unit instanceof VideoBlockModel &&
                     ((VideoBlockModel) unit).getData().encodedVideos.getPreferredVideoInfo() != null) {
-                    unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit);
+                unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit);
             } else if (unit instanceof VideoBlockModel &&
                     ((VideoBlockModel) unit).getData().encodedVideos.getYoutubeVideoInfo() != null) {
-                    unitFragment = CourseUnitOnlyOnYoutubeFragment.newInstance(unit);
+                unitFragment = CourseUnitOnlyOnYoutubeFragment.newInstance(unit);
+            } else if (unit.getType() == BlockType.DISCUSSION) {
```

Comment:
This should be changed to `unit instanceof DiscussionBlockModel`, since the parser checks the availability of the fields as well in order to create this model.